resolve_attr_refs.go

```
@@ -0,0 +1,59 @@
+package gowsdl
+
+import (
+	"strings"
+)
+
+// resolveAttrRefs resolves all attribute references across the provided schemas.
+// It modifies the schemas in-place, copying properties from the referenced attributes
+// onto the referencing attributes.
+func resolveAttrRefs(schemas []*XSDSchema) {
+	// First, build an index from (namespace, attrName) -> attrDef
+	// for all global attrs across all schemas.
+	attrIndex := make(map[namespacedKey]*XSDAttribute)
+	for _, s := range schemas {
+		for _, attr := range s.Attributes {
+			if attr.Name != "" {
+				attrIndex[newNamespacedKey(s.TargetNamespace, attr.Name)] = attr
+			}
+		}
+	}
+
+	keyFromAttrRef := func(s *XSDSchema, attrRef string) namespacedKey {
+		parts := strings.SplitN(attrRef, ":", 2)
+		if len(parts) == 1 {
+			return newNamespacedKey(s.TargetNamespace, parts[0])
+		} else {
+			if ns, ok := s.Xmlns[parts[0]]; ok {
+				return newNamespacedKey(ns, parts[1])
+			}
+		}
+		return ""
```

It seems like we'll hit this if `parts[0]` isn't in Xmlns, which seems pretty unexpected, no? Should this fail louder?

---

Probably, yes. It was a silent failure before, so I left it as such, but I'll update.

---

047cf4c16bd80f0b0f248f3ddcc0e28b9186fe85